pkg/providers/internal/openstack/provider.go

```diff
@@ -532,8 +545,55 @@ func getPublicOrOrganizationOwnedImages(resources []images.Image, organizationID
 	return result
 }
 
+func (p *Provider) imageDiskFormat(image *images.Image) types.ImageDiskFormat {
+	return types.ImageDiskFormat(image.DiskFormat)
+}
+
+func (p *Provider) imageDataSource(image *images.Image) types.ImageDataSource {
+	temp, _ := image.Properties["unikorn:data_source"].(string)
+
+	var dataSource types.ImageDataSource
```

Comment:
Eyes hurt... 
```go
dataSource := types.ImageDataSource(temp)

if dataSource == types.ImageDataSourceFile || dataSource == types.ImageDataSourceURL) {
  return dataSource
}

return types.ImageDataSourceFile
```

Reply:
Not sure what we do if it's something _other_ than these consts